src/actions/profile-view.js

```diff
@@ -216,16 +218,58 @@ export function isolateGlobalTrack(
     sendAnalytics({
       hitType: 'event',
       eventCategory: 'timeline',
-      eventAction: 'isolate global track',
+      eventAction: 'isolate process',
+    });
+
+    dispatch({
+      type: 'ISOLATE_PROCESS',
+      hiddenGlobalTracks: new Set(
+        trackIndexes.filter(i => i !== isolatedTrackIndex)
+      ),
+      isolatedTrackIndex,
+      selectedThreadIndex,
+    });
+  };
+}
+
+/**
+ * This function isolates a global track, and hides all of its local tracks.
+ */
+export function isolateProcessMainThread(
+  isolatedTrackIndex: TrackIndex
+): ThunkAction<void> {
+  return (dispatch, getState) => {
+    const track = getGlobalTracks(getState())[isolatedTrackIndex];
+    const trackIndexes = getGlobalTrackOrder(getState());
+
+    if (track.type !== 'process') {
+      // Do not isolate a track unless it is a process track.
+      return;
+    }
+
+    const selectedThreadIndex = track.mainThreadIndex;
+    if (selectedThreadIndex === null) {
+      // Make sure that a thread really exists.
+      return;
+    }
+
+    sendAnalytics({
```

Comment:
Just curious, why do we gather analytics information here?

Reply:
When I originally implemented the analytics, I tried to get coverage over all the main user-facing actions, to see how and if users used things. It turns out people don't really re-organize threads, but they do hide them.

Reply:
Interesting. Thanks :)